pandas/_libs/src/klib/khash_python.h

```diff
@@ -163,18 +163,81 @@ KHASH_MAP_INIT_COMPLEX128(complex128, size_t)
 #define kh_exist_complex128(h, k) (kh_exist(h, k))
 
 
+int PANDAS_INLINE floatobject_cmp(PyFloatObject* a, PyFloatObject* b){
```

Comment:
can you add some comments before these of what is going on / why doing is & the issue reference